File: CredentialsExtractor/Cryptography/AesCryptographyService.cs
// AesCryptographyService.cs
using System;
using System.IO;
using System.Security.Cryptography;
using System.Text;

namespace CredentialsExtractor.Cryptography
{
    public class AesCryptographyService : ICryptographyService
    {
        private readonly byte[] _key;
        private readonly byte[] _iv;

        public AesCryptographyService(string base64Key = null, string base64Iv = null)
        {
            // In production, these should be securely exchanged, not hardcoded
            // Using base64 encoded 32-byte key (256-bit) and 16-byte IV (128-bit)
            _key = Convert.FromBase64String(base64Key ?? "dGhpc2lzYXZlcnlzZWN1cmVrZXkxMjM0NTY3ODkwMTI=");
            _iv = Convert.FromBase64String(base64Iv ?? "dGhpc2lzMTZieXRlc2l2IQ==");
        }

        public string Encrypt(string plainText)
        {
            if (string.IsNullOrEmpty(plainText))
                return string.Empty;

            byte[] encrypted = EncryptBytes(Encoding.UTF8.GetBytes(plainText));
            return Convert.ToBase64String(encrypted);
        }

        public string Decrypt(string cipherText)
        {
            if (string.IsNullOrEmpty(cipherText))
                return string.Empty;

            byte[] cipherBytes = Convert.FromBase64String(cipherText);
            byte[] decrypted = DecryptBytes(cipherBytes);
            return Encoding.UTF8.GetString(decrypted);
        }

        public byte[] EncryptBytes(byte[] plainBytes)
        {
            if (plainBytes == null || plainBytes.Length == 0)
                return new byte[0];

            using (Aes aes = Aes.Create())
            {
                aes.Key = _key;
                aes.IV = _iv;
                aes.Mode = CipherMode.CBC;
                aes.Padding = PaddingMode.PKCS7;

                using (var encryptor = aes.CreateEncryptor())
                using (var msEncrypt = new MemoryStream())
                {
                    using (var csEncrypt = new CryptoStream(msEncrypt, encryptor, CryptoStreamMode.Write))
                    {
                        csEncrypt.Write(plainBytes, 0, plainBytes.Length);
                        csEncrypt.FlushFinalBlock();
                    }
                    return msEncrypt.ToArray();
                }
            }
        }

        public byte[] DecryptBytes(byte[] cipherBytes)
        {
            if (cipherBytes == null || cipherBytes.Length == 0)
                return new byte[0];

            using (Aes aes = Aes.Create())
            {
                aes.Key = _key;
                aes.IV = _iv;
                aes.Mode = CipherMode.CBC;
                aes.Padding = PaddingMode.PKCS7;

                using (var decryptor = aes.CreateDecryptor())
                using (var msDecrypt = new MemoryStream(cipherBytes))
                using (var csDecrypt = new CryptoStream(msDecrypt, decryptor, CryptoStreamMode.Read))
                using (var msPlain = new MemoryStream())
                {
                    csDecrypt.CopyTo(msPlain);
                    return msPlain.ToArray();
                }
            }
        }
    }

    // Factory for creating cryptography services
    public class CryptographyServiceFactory
    {
        public static ICryptographyService CreateDefault()
        {
            return new AesCryptographyService();
        }

        public static ICryptographyService CreateWithKeys(string base64Key, string base64Iv)
        {
            return new AesCryptographyService(base64Key, base64Iv);
        }
    }
}

File: CredentialServer/cryptography/aes_cryptography_service.py
"""aes_cryptography_service.py"""
import base64
from Crypto.Cipher import AES
from Crypto.Util.Padding import pad, unpad
import sys
import os
sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
from interfaces.cryptography_service import ICryptographyService

class AESCryptographyService(ICryptographyService):
    """AES encryption implementation using CBC mode"""
    
    def __init__(self, base64_key: str = None, base64_iv: str = None):
        """Initialize with base64 encoded key and IV"""
        # Default keys should match the C# implementation
        self.key = base64.b64decode(base64_key or "dGhpc2lzYXZlcnlzZWN1cmVrZXkxMjM0NTY3ODkwMTI=")
        self.iv = base64.b64decode(base64_iv or "dGhpc2lzMTZieXRlc2l2IQ==")
    
    def encrypt(self, plain_text: str) -> str:
        """Encrypt plain text and return base64 encoded cipher text"""
        if not plain_text:
            return ""
        
        plain_bytes = plain_text.encode('utf-8')
        encrypted_bytes = self.encrypt_bytes(plain_bytes)
        return base64.b64encode(encrypted_bytes).decode('utf-8')
    
    def decrypt(self, cipher_text: str) -> str:
        """Decrypt base64 encoded cipher text and return plain text"""
        if not cipher_text:
            return ""
        
        cipher_bytes = base64.b64decode(cipher_text)
        decrypted_bytes = self.decrypt_bytes(cipher_bytes)
        return decrypted_bytes.decode('utf-8')
    
    def encrypt_bytes(self, plain_bytes: bytes) -> bytes:
        """Encrypt raw bytes using AES-CBC with PKCS7 padding"""
        if not plain_bytes:
            return b""
        
        # Create cipher
        cipher = AES.new(self.key, AES.MODE_CBC, self.iv)
        
        # Pad data using PKCS7
        padded_data = pad(plain_bytes, AES.block_size)
        
        # Encrypt
        encrypted = cipher.encrypt(padded_data)
        
        return encrypted
    
    def decrypt_bytes(self, cipher_bytes: bytes) -> bytes:
        """Decrypt raw bytes using AES-CBC with PKCS7 padding"""
        if not cipher_bytes:
            return b""
        
        # Create cipher
        cipher = AES.new(self.key, AES.MODE_CBC, self.iv)
        
        # Decrypt
        decrypted_padded = cipher.decrypt(cipher_bytes)
        
        # Remove padding
        decrypted = unpad(decrypted_padded, AES.block_size)
        
        return decrypted